src/piptool.py

```diff
@@ -62,18 +62,19 @@ def as_tuple(preq):
     return name, version, extras
 
 
-def repository_name(name, version, python_version):
+def repository_name(repo_prefix, name, version, python_version):
     """Returns the canonical name of the Bazel repository for a package.
 
-    :param name: package nane
-    :param version: package verion
+    :param repo_prefix: prefix to attach to the repo
+    :param name: package name
+    :param version: package version
     :param python_version: python major version
     :returns: repo name
     :rtype: str
 
     """
-    canonical = "pypi__{}__{}_{}".format(python_version, name, version)
-    return clean_name(canonical)
+    canonical = "__{}__{}_{}".format(python_version, name, version)
+    return f"{repo_prefix}{clean_name(canonical)}"
```

Comment:
`repo_prefix` is deliberately left out of sanitisation cause it passes through as the user wants it. This was the only way I can get around stuff like python packages looking for `site-packages`.
If there is a better idea I am happy to be corrected.

Reply:
TODO: Make this python2 compatible.